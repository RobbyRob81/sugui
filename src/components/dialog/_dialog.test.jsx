import * as React from 'react';
import { customRender } from '../../../test/test-utils';
import Dialog from '.';

it('should render without throwing an error', () => {
  const { container } = customRender(<Dialog>Test</Dialog>);
  expect(container.firstChild).toMatchSnapshot();
});

it('should hide content by default', async () => {
  const { getByText } = customRender(<Dialog><p>Test</p></Dialog>);
  expect(getByText('Test')).not.toBeVisible();
});

it('should display content if opened is set', () => {
  const { getByText } = customRender(<Dialog open><p>Test</p></Dialog>);
  expect(getByText('Test')).toBeVisible();
});

it('should display a heading', () => {
  const { getByText } = customRender(<Dialog open heading="hello"><p>Test</p></Dialog>);
  expect(getByText('hello')).toBeVisible();
});
