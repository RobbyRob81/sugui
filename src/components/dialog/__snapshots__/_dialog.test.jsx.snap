// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render without throwing an error 1`] = `
<div
  class="sc-bwzfXH ifBdPs"
>
  <section
    class="sc-htpNat kkWFOy sc-bdVaJa dWiwem"
  >
    <h2
      class="sc-ifAKCX bpJlYg"
    />
    <div
      class="sc-bxivhb cLjUpK"
    >
      Test
    </div>
  </section>
</div>
`;
